### PR TITLE
feat: customize space and list icons

### LINKED ISSUE #125: Allow user to select space/list icons with BaseIconSelector
- User should be able to set icon and color
- We should use the same icon for the list we use in BaseList component in the navigation bar next to list and space names instead of existing icons

### PR DESCRIPTION
**Pull request by Us3r-gitHub**
Closes #125

### CODE REVIEW

File: packages/backend/src/migrations/1720414956770-add-icon-column-to-space-and-list-tables.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { MigrationInterface, QueryRunner, TableColumn } from "typeorm";
+
+const iconColumnProperties = {
+    name: "icon",
+    type: "varchar",
+    length: "255",
+    isNullable: true,
```
Comment (MoShizzle) [inline packages/backend/src/migrations/1720414956770-add-icon-column-to-space-and-list-tables.ts:7]: Remove isNullable and set `mdi-list-box-outline` for lists and `mdi-folder-outline` for spaces as default

**Review by MoShizzle — CHANGES_REQUESTED** [review]
We should also be able to select icon color with BaseColorPicker along with the icon, for both lists and spaces

File: packages/backend/src/migrations/1720414956770-add-icon-column-to-space-and-list-tables.ts (hunk 1 above)
Comment (Us3r-gitHub) [reply packages/backend/src/migrations/1720414956770-add-icon-column-to-space-and-list-tables.ts:7]: Should it be define in migration?
Or in entity file?

Comment (Us3r-gitHub) [reply packages/backend/src/migrations/1720414956770-add-icon-column-to-space-and-list-tables.ts]: Added on [93102bc](https://github.com/tillywork/tillywork/pull/133/commits/93102bc6f4c908c7b545fc7364da6fa3634418bb)

**Review by MoShizzle — APPROVED** [review]

**Pull request merged**